### PR TITLE
Support one handler for mulitple GCP services and fixes

### PR DESCRIPTION
**Pull request by manlinl**
- Allow one handler handles multiple GCP services
- Expand metric support
- Minor fixes in reportBuilder

### CODE REVIEW

File: mixer/adapter/svcctrl/handler.go [hunk 1]
```diff
@@ -97,39 +102,76 @@ func newServiceProcessor(meshServiceName string, ctx *handlerContext) (*serviceP
...
 }
 
 // HandleQuota handles rate limiting quota.
 func (h *handler) HandleQuota(ctx context.Context, instance *quota.Instance,
 	args adapter.QuotaArgs) (adapter.QuotaResult, error) {
-	return h.svcProc.ProcessQuota(ctx, instance, args)
+	svcProc, err := h.getServiceProcessor(ctx)
+	if err != nil {
+		return adapter.QuotaResult{
+			Status: status.WithPermissionDenied(err.Error()),
```
Comment (apicl) [inline mixer/adapter/svcctrl/handler.go]: Is this internal error? It does not look like a permission issue.

File: mixer/adapter/svcctrl/handler.go [hunk 2]
```diff
@@ -67,9 +71,10 @@ type (
 
 	handler struct {
 		ctx *handlerContext
-		// TODO(manlinl): Switch to a LRU cache of serviceProcessor once Mixer includes destination.service in all
-		// instances by default. Then we can enable a single handler to server multiple services.
-		svcProc *serviceProcessor
+
+		// lock protects svcProcMap.
+		lock       sync.Mutex
+		svcProcMap map[string]*serviceProcessor
```
Comment (apicl) [inline mixer/adapter/svcctrl/handler.go]: Please comment what's the key/value?

File: mixer/adapter/svcctrl/handler.go [hunk 3]
```diff
@@ -97,39 +102,76 @@ func newServiceProcessor(meshServiceName string, ctx *handlerContext) (*serviceP
...
 		return nil, err
 	}
-	return &handler{ctx, svcProc}, nil
+	h.svcProcMap[serviceFullName] = svcProc
+	return svcProc, nil
+}
+
+func newHandler(ctx *handlerContext) (*handler, error) {
+	return &handler{
+		ctx:        ctx,
```
Comment (apicl) [inline mixer/adapter/svcctrl/handler.go]: Please test this code.

File: mixer/adapter/svcctrl/handler.go (hunk 2 above)
Comment (manlinl) [reply mixer/adapter/svcctrl/handler.go]: done

File: mixer/adapter/svcctrl/handler.go (hunk 1 above)
Comment (manlinl) [reply mixer/adapter/svcctrl/handler.go]: fixed

File: mixer/adapter/svcctrl/handler.go (hunk 3 above)
Comment (manlinl) [reply mixer/adapter/svcctrl/handler.go]: done.

**Pull request merged**